src/main/java/com/ays/backend/user/service/RoleService.java

```diff
@@ -0,0 +1,93 @@
+package com.ays.backend.user.service;
+
+import com.ays.backend.user.exception.RoleNotFoundException;
+import com.ays.backend.user.model.ERole;
+import com.ays.backend.user.model.Role;
+import com.ays.backend.user.repository.RoleRepository;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.HashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class RoleService {
+
+    private final RoleRepository roleRepository;
+
+    public Optional<Role> findByName(ERole name) {
+        return roleRepository.findByName(name);
+    }
+
+    public Set<Role> addRoleToUser(Set<String> strRoles ) {
```

Comment:
please make sure to decrease the complexity here.

Reply:
I just revised this method